R/evaluate_projection.r

```diff
@@ -0,0 +1,215 @@
+#' Evaluate a projection model
+#'
+#' To evaluate how successfully a projection model predicts future uptake,
+#' plots of predicted incident and cumulative uptake are generated, along with
+#' the data against which the model is being compared. The model ought to
+#' match the data closely, although uncertainty may propagate through time.
+#' Each plot shows the data and model predictions with credible intervals
+#' (95% by default).
+#'
+#' If only old data, i.e. the data on which the model was fit, is provided,
+#' the projections will be compared against that. If new data, i.e. data
+#' which the model has not seen, is also provided, projections will compared
+#' against that instead. In the latter case, the fit may not be as good, and
+#' credible intervals will be wider due to extra uncertainty over the values
+#' of year-specific parameters.
+#'
+#' If no start date is provided, it is assumed the exact incident and
+#' cumulative uptake is known for each time t-1 when incident uptake for
+#' time t is being predicted. This evaluates how well the model fits
+#' each individual data point through the entire time series.
```

Comment:
I'm not sure if I understand this. How can we predict without a start date?

Reply:
If no start date is given, each data point is predicted as if the previous data point were the start date. In other words, each data point is predicted as if the previous data point is known. This is useful for diagnosing the model on its training data, not for making bona fide projections of the future.